.jenkins

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+echo "*** $(basename $0)"
+set -e
+set -x
+
+ARTIFACT=$(gem build *.gemspec | grep File | awk '{print $2}')
+
+if [[ "$PUBLISH" == "1" ]]; then
```

Comment:
@L2G 

Make sure this is not going to be an issue as I am sure you don't want people uploading random crap ;)

The rest of the commits look fine, no worries.